Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -419,9 +419,13 @@ void TextureMapperLayer::collectDamageSelf(TextureMapperPaintOptions& options, D
     transform.multiply(options.transform);
     transform.multiply(m_layerTransforms.combined);
 
-    if (m_damage.isInvalid())
-        damage.invalidate();
-    else if (m_contentsLayer && m_damage.isEmpty()) {
+    ASSERT(!m_damage.isInvalid() && !m_inferredDamage.isInvalid());
```

Comment:
Assertions are really helpful, if there’s only one condition checked. Please separate in two.